airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx

```diff
@@ -68,10 +73,57 @@ export const AuthenticationProvider: React.FC = ({ children }) => {
   const analytics = useAnalyticsService();
   const authService = useInitService(() => new GoogleAuthService(() => auth), [auth]);
 
+  /**
+   * Create a user object in the Airbyte database from an existing Firebase user.
+   * This will make sure that the user account is tracked in our database as well
+   * as create a workspace for that user. This method also takes care of sending
+   * the relevant user creation analytics events.
+   */
+  const createDatabaseUser = async (
+    fbUser: FbUser,
+    userData: { name?: string; companyName?: string; news?: boolean } = {}
+  ): Promise<User> => {
+    // Create the Airbyte user on our server
+    const user = await userService.create({
+      authProvider: AuthProviders.GoogleIdentityPlatform,
+      authUserId: fbUser.uid,
+      email: fbUser.email ?? "",
+      name: userData.name ?? fbUser.displayName ?? "",
+      companyName: userData.companyName ?? "",
+      news: userData.news ?? false,
+    });
+
+    analytics.track(Namespace.USER, Action.CREATE, {
+      actionDescription: "New user registered",
+      user_id: fbUser.uid,
+      name: user.name,
+      email: user.email,
+      // Which login provider was used, e.g. "password", "google.com", "github.com"
+      provider: fbUser.providerData[0]?.providerId,
```

Comment:
:information_source: There is a chance a firebase user has multiple providers linked (e.g. mail and google as mentioned in the PR description). But that shouldn't be the case when the user is just created, in which case they should only have one provider linked, thus we can simply take the first provider here for sending the event.